processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -327,23 +322,57 @@ public void writeToChannel(WritableByteChannel channel) throws IOException
   }
 
   /**
-   * The returned GenericIndexed must be closed to release the underlying memory
+   * Returns an Indexed wrapping the underlying indexed with an optional value lookup cache
+   * if supported by the underlying ObjectStrategy.
+   *
+   * The indexed that is returned by this method:
+   *   1) is not thread safe and should only be used by a single thread at a time.
+   *   2) must be closed to release any resources, regardless of the specified cache size.
    *
-   * @param maxBytes maximum size in bytes of the lookup cache
-   * @return a copy of this GenericIndexed with a lookup cache.
+   * A caching indexed may perform better than the underlying Indexex even without a value cache
+   * since it does not have to guarantee thread-safety.
+   *
+   * @param lookupCacheSize maximum size in bytes of the lookup cache if greater than zero
+   * @return an Indexed wrapping the underlying indexed with a lookup cache.
    */
-  public GenericIndexed<T> withCache(int maxBytes)
+  public CloseableIndexed<T> getCachingIndexed(final int lookupCacheSize)
```

Comment:
will we ever know if a closeable indexed isn't closed somewhere or can another leak occur? I guess I'm wondering if there's a way to check if resources are being closed and throw an exception if they are not.